aten/src/ATen/native/cuda/TensorShape.cu

```diff
@@ -428,4 +707,69 @@ void split_with_sizes_copy_out_cuda(
   }
 }
 
+Tensor _chunk_cat_cuda(TensorList tensors, int64_t dim, int64_t num_chunks) {
+  dim = at::native::preprocess_chunk_cat_inputs(tensors, dim, num_chunks);
+  if (detail::all_contiguous(tensors)) {
+    // Return a tensor with the same dtype as input tensors
+    int64_t elem_size = tensors[0].element_size();
+    auto
+        [chunk_size,
+         leading_dim,
+         num_blocks_per_chunk,
+         slice_size,
+         device_ptrs] =
+            detail::get_chunk_cat_metadata(
+                tensors, dim, num_chunks, elem_size, elem_size);
+    auto view_sizes = detail::get_chunk_cat_out_sizes(
+        tensors[0].sizes(), dim, num_chunks, chunk_size, elem_size);
+    Tensor out =
+        tensors[0]
+            .new_empty(chunk_size * num_chunks * leading_dim / elem_size)
+            .view(view_sizes);
+    // Type-agnostic copy since out and input tensors have the same type.
+    detail::_chunk_cat_out_cuda_contiguous<char, char>(
+        tensors, dim, num_chunks, out, elem_size, elem_size);
+    return out;
+  } else {
+    return at::native::_chunk_cat(tensors, dim, num_chunks);
+  }
+}
+
+Tensor& _chunk_cat_out_cuda(
+    TensorList tensors,
+    int64_t dim,
+    int64_t num_chunks,
+    Tensor& out) {
+  dim = at::native::preprocess_chunk_cat_inputs(tensors, dim, num_chunks);
+  TORCH_CHECK(
+      tensors[0].device() == out.device(),
+      "_chunk_cat_out_cuda: mismatch between input and out tensor devices");
+  bool both_input_output_contiguous =
+      detail::all_contiguous(tensors) && out.is_non_overlapping_and_dense();
+  if (both_input_output_contiguous &&
+      (tensors[0].dtype() == at::ScalarType::BFloat16) &&
+      (out.dtype() == at::ScalarType::Float)) {
+    detail::_chunk_cat_out_cuda_contiguous<float, BFloat16>(
```

Comment:
Maybe comment here that `_chunk_cat_out_cuda_contiguous` works for other type. Not dispatching for other types because this configuration is the only known use case.